packages/main/src/Menu.ts

```diff
@@ -124,8 +124,15 @@ type OpenerStandardListItem = StandardListItem & { associatedItem: MenuItem };
  * @event sap.ui.webc.main.Menu#before-open
  * @allowPreventDefault
  * @since 1.10.0
+ * @param {object} item The <code>ui5-menu-item</code> that triggers opening of the sub-menu. Note: available since 1.14.0.
  */
-@event("before-open")
+@event("before-open", {
+	detail: {
+		item: {
+			type: Object,
```

Comment:
if the ui5-menu-item is the item in the details, type: HTMLElement is more appropriate